src/coreclr/src/vm/virtualcallstub.cpp

```diff
@@ -1914,7 +1914,11 @@ PCODE VirtualCallStubManager::ResolveWorker(StubCallSite* pCallSite,
                         {
                             pResolveHolder = GenerateResolveStub(pResolverFcn,
                                                              pBackPatchFcn,
-                                                             token.To_SIZE_T());
+                                                             token.To_SIZE_T()
+#ifdef TARGET_X86
+                                                             , objectType
```

Comment:
Does the `objectType` have the right lifetime? Can the resolve stub outlive it?

(Look for `bCallToShorterLivedTarget` above for some context.)

Reply:
Thank you for pointing it out, it seems it can happen. It seems that I can fix that by storing the stack arguments size in the stub instead of the MethodTable for the resolve stub.

Reply:
I've just pushed a commit that replaces the usage of MethodTable by the size of stack arguments in the ResolveStub.